core/play/src/main/java/play/mvc/Results.java

```
@@ -4537,6 +4545,51 @@ public static Result redirect(String url) {
     return new Result(SEE_OTHER, Collections.singletonMap(LOCATION, url));
   }
 
+  /**
+   * Generates a 303 See Other result.
+   *
+   * @param url The url to redirect
+   * @param queryStringParams queryString parameters to add to the queryString
+   * @return the result
+   */
+  public static Result redirect(String url, Map<String, List<String>> queryStringParams) {
+    String fullUrl = addQueryStringParams(url, queryStringParams);
+    return new Result(SEE_OTHER, Collections.singletonMap(LOCATION, fullUrl));
+  }
+
+  /**
+   * Encodes and adds the query params to the given url
+   *
+   * @param url
+   * @param queryStringParams
+   * @return
+   */
+  public static String addQueryStringParams(
+      String url, Map<String, List<String>> queryStringParams) {
+    if (queryStringParams.isEmpty()) {
+      return url;
+    } else {
+      String queryString =
+          queryStringParams.entrySet().stream()
+              .flatMap(
+                  entry ->
+                      entry.getValue().stream()
+                          .map(parameterValue -> encodeParameter(entry.getKey(), parameterValue)))
+              .collect(Collectors.joining("&"));
+
+      return url + (url.contains("?") ? "&" : "?") + queryString;
+    }
+  }
+
+  private static String encodeParameter(String key, String value) {
+    try {
+      return URLEncoder.encode(key, "utf-8") + "=" + URLEncoder.encode(value, "utf-8");
+    } catch (UnsupportedEncodingException e) {
+      logger.error("failed to encode, unsupported encoding!", e);
+      return key + "=" + value;
+    }
+  }
```

This is changing behavior (swallows the exception).

---

resolving this since the implementation is moved to scala and this method is removed